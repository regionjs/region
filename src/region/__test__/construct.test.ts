import { Region } from '../..';

describe('constructor', () => {
  test('with config', () => {
    const userRegion = new Region({ name: 'user' });
    expect(userRegion).toMatchSnapshot();
  });

  test('with string', () => {
    const followerRegion = new Region('follower');
    expect(followerRegion).toMatchSnapshot();
  });

  test('with undefined', () => {
    // @ts-ignore
    const globalRegion = new Region();
    expect(globalRegion).toMatchSnapshot();
  });
});
