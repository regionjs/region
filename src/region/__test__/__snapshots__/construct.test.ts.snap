// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constructor with config 1`] = `
RegionReact {
  "connect": [Function],
  "connectWith": [Function],
  "enableLog": true,
  "expiredTime": 0,
  "getProps": [Function],
  "load": [Function],
  "loadBy": [Function],
  "name": "user",
  "private_actionTypes": Object {
    "LOAD": "@user/LOAD",
    "RESET": "@user/RESET",
    "SET": "@user/SET",
  },
  "private_getErrors": [Function],
  "private_getFetchTimes": [Function],
  "private_getLoadings": [Function],
  "private_getResults": [Function],
  "private_getState": [Function],
  "private_reducer": [Function],
  "private_setConfig": [Function],
  "private_store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "reset": [Function],
  "set": [Function],
  "setBy": [Function],
  "strictLoading": true,
  "useProps": [Function],
}
`;

exports[`constructor with string 1`] = `
RegionReact {
  "connect": [Function],
  "connectWith": [Function],
  "enableLog": true,
  "expiredTime": 0,
  "getProps": [Function],
  "load": [Function],
  "loadBy": [Function],
  "name": "follower",
  "private_actionTypes": Object {
    "LOAD": "@follower/LOAD",
    "RESET": "@follower/RESET",
    "SET": "@follower/SET",
  },
  "private_getErrors": [Function],
  "private_getFetchTimes": [Function],
  "private_getLoadings": [Function],
  "private_getResults": [Function],
  "private_getState": [Function],
  "private_reducer": [Function],
  "private_setConfig": [Function],
  "private_store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "reset": [Function],
  "set": [Function],
  "setBy": [Function],
  "strictLoading": true,
  "useProps": [Function],
}
`;

exports[`constructor with undefined 1`] = `
RegionReact {
  "connect": [Function],
  "connectWith": [Function],
  "enableLog": true,
  "expiredTime": 0,
  "getProps": [Function],
  "load": [Function],
  "loadBy": [Function],
  "name": "_",
  "private_actionTypes": Object {
    "LOAD": "@_/LOAD",
    "RESET": "@_/RESET",
    "SET": "@_/SET",
  },
  "private_getErrors": [Function],
  "private_getFetchTimes": [Function],
  "private_getLoadings": [Function],
  "private_getResults": [Function],
  "private_getState": [Function],
  "private_reducer": [Function],
  "private_setConfig": [Function],
  "private_store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "reset": [Function],
  "set": [Function],
  "setBy": [Function],
  "strictLoading": true,
  "useProps": [Function],
}
`;
